packages/desktop/views/onboarding/views/complete-onboarding/views/FinishOnboardingView.svelte

```diff
@@ -1,12 +1,27 @@
 <script lang="ts">
     import { OnboardingLayout } from '@components'
-    import { completeOnboardingProcess, isOnboardingLedgerProfile } from '@contexts/onboarding'
+    import {
+        RestoreProfileType,
+        completeOnboardingProcess,
+        isOnboardingLedgerProfile,
+        onboardingProfile,
+    } from '@contexts/onboarding'
     import { mobile } from '@core/app'
     import { localize } from '@core/i18n'
     import { checkOrConnectLedger } from '@core/ledger'
+    import { isStrongholdUnlocked, setStrongholdPassword } from '@core/profile-manager'
     import { Animation, Button, Icon, Text, TextHint } from '@ui'
     import { onboardingRouter } from '@views/onboarding/onboarding-router'
 
+    async function verifyAndRestoreStronghold(): Promise<void> {
+        if ($onboardingProfile?.restoreProfileType === RestoreProfileType.Stronghold) {
+            const _isStrongholdUnlocked = await isStrongholdUnlocked()
```

Comment:
to simplify and also prevent possible race conditions, i think we are safe by always unlocking, without checking